### PR TITLE
clear up max_size docs and have code match the docs

### LINKED ISSUE #781: max_size is ambiguously defined in the docs
[The check to see if the SVG is too large](https://github.com/mapnik/node-mapnik/blob/fe80ce5d79c0e90cfbb5a2b992bf0ae2b8f88198/src/mapnik_image.cpp#L2687-L2693) checks `width` and `height` separately while the documentation states that `max_size` is [the maximum allowed size of the svg dimensions * scale](https://github.com/mapnik/node-mapnik/blob/fe80ce5d79c0e90cfbb5a2b992bf0ae2b8f88198/src/mapnik_image.cpp#L2549-L2551).

Reading this over a few times, I think this is a correct statement, but on first pass the docs seemed to imply area `width * height * scale^2` while the code enforces longest side `Math.max(width, height) * scale`.

As the goal of this check is to make sure too much memory is not allocated for an Image, it seems that the area of the image is what we should be enforcing here. Does that sound right @flippmoke and @springmeyer?

### PR DESCRIPTION
**Pull request by mcwhittemore**
Closes #781

### CODE REVIEW

File: src/mapnik_image.cpp [hunk 1]
```diff
@@ -2546,7 +2546,7 @@ NAN_METHOD(Image::fromSVGBytesSync)
  * @param {Object} [options]
  * @param {number} [options.scale] - scale the image. For example passing `0.5` as scale would render
  * your SVG at 50% the original size.
- * @param {number} [options.max_size] - the maximum allowed size of the svg dimensions * scale. The default is 2048.
+ * @param {number} [options.max_size] - the maximum allowed size of the svg area at the provided scale. The default is 4194304.
```
Comment (mcwhittemore) [inline src/mapnik_image.cpp]: I've changed the default min from 2048 to 4194304 (2048^2) as 2048 was the max size of an edge before this change. This should mean that images that use to pass by default, will still pass.

**Comment by mcwhittemore** [comment]
Working through test failures, it looks like the max value for `max_size` is also based of the size of one dimension. Should [this] be squared as well?

**Comment by davenquinn** [comment]
Just upgraded my node-mapnik to the `3.0.x` branch and am getting a `max_size` related error. Namely, for some large maps rendered with the `Map.render` or `Map.renderSync`, I get an `Error: Invalid width for image dimensions requested`. These maps rendered perfectly fine previously. I'm not sure how to address. Is there a way to globally set the `max_size` constraint because that would likely be simplest.

**Comment by springmeyer** [comment]
@davenquinn my intuition is that your issue is different that what this PR is attempting to fix. Your error is coming from mapnik core: https://github.com/mapnik/mapnik/blob/v3.0.x/include/mapnik/image_impl.hpp and indicates that an image larger than `65535x65535` is being allocated (https://github.com/mapnik/mapnik/blob/v3.0.x/include/mapnik/image.hpp#L77).

**Comment by davenquinn** [comment]
OK thanks, that's really big, much larger than the image I'm trying to produce. I'll look in the mapnik 3.0.13 > 3.0.16 transition to see what might have caused that in core.

**Comment by springmeyer** [comment]
This is stale and looking back at this I'm not totally convinced its a necessary change. @mcwhittemore if you feel otherwise feel free to re-open a new ticket, but for now I'll close this one.

**Pull request closed without merging**